File: graph.lua
function addvertex(value, graph) 
  graph[value] = {}
end

function addedge(v1, v2, graph)
  if graph[v1] == nil or graph[v2] == nil then
    print("those values don't exist")
    return
  end
  table.insert(graph[v1], v2)
  table.insert(graph[v2], v1)
end

function printConnections(graph)
  for key, value in pairs(graph) do
    print("Values for: " .. key)
    for i=1, #value do
      print(value[i])
    end
  end
end

local graph = {}
addvertex("blitz", graph)
addvertex("bob", graph)
addvertex("john", graph)
addedge("blitz", "bob", graph)
addedge("blitz", "john", graph)
printConnections(graph)

File: graph.go
package main

import (
	"fmt"
)

type Graph struct {
	data map[string][]string
}

func addVertex(value string, graph *Graph) {
	graph.data[value] = make([]string, 10)
}

func addEdge(v1 string, v2 string, graph *Graph) {
	if graph.data[v1] == nil || graph.data[v2] == nil {
		fmt.Println("Not valid keys")
		return 
	}
	graph.data[v1] = append(graph.data[v1], v2)
	graph.data[v2] = append(graph.data[v2], v1)
}

func get(value string, graph *Graph) []string {
	return graph.data[value]
}

func main() {
	graph := Graph {
		data: make(map[string][]string),
	}
	addVertex("Blitz", &graph)
	addVertex("John", &graph)
	addVertex("Justin", &graph)
	addEdge("Blitz", "John", &graph)
	addEdge("Justin", "Blitz", &graph)
	
	connections := get("Blitz", &graph)
	for i := 0; i < len(connections); i++ {
		if connections[i] != "" {
			fmt.Println(connections[i])
		}
	}
}
